### ISSUE TITLE
zpool status -v -c XXX,YYY,ZZZ does not populate the requested fields

### ISSUE THREAD

**Issue by stemsin** [openzfs/zfs#13404]
### System information

Type | Version/Name
--- | ---
Distribution Name | TrueNAS Core
Distribution Version | 12.2-RELEASE-p12
Architecture | amd64
OpenZFS Version | zfs-2.0.7-1, zfs-kmod-v2022012000-zfs_e870bd8cf

### Describe the problem you're observing
Running the following command
```
ZPOOL_SCRIPTS_AS_ROOT=1 zpool status -v -c health,model,hours_on,dmdeps,enc,slot,temp,lsblk,defect,size,model,media,test_status,encdev,label,serial
```

### Actual result
The additional fields requested are blank
```
pool: vault_sas
state: ONLINE
scan: scrub repaired 1.89M in 01:53:33 with 0 errors on Sun Apr 17 01:53:35 2022
config:

NAME                                            STATE    READ WRITE CKSUM  health  model  hours_on  dm-deps  enc  slot  temp  size  vendor  defect  media  test_status  encdev  label  serial
vault_sas                                      ONLINE      0    0    0
mirror-0                                      ONLINE      0    0    0
gptid/1dc1d415-14c4-11ec-b389-10604b9b4de0  ONLINE      0    0    0      -      -        -        -    -    -    -    -      -      -      -            -      -      -      -
gptid/1e28374a-14c4-11ec-b389-10604b9b4de0  ONLINE      0    0    0      -      -        -        -    -    -    -    -      -      -      -            -      -      -      -
mirror-1                                      ONLINE      0    0    0
gptid/1e20b28a-14c4-11ec-b389-10604b9b4de0  ONLINE      0    0    0      -      -        -        -    -    -    -    -      -      -      -            -      -      -      -
gptid/1e38f486-14c4-11ec-b389-10604b9b4de0  ONLINE      0    0    0      -      -        -        -    -    -    -    -      -      -      -            -      -      -      -
mirror-2                                      ONLINE      0    0    0
gptid/4f166f8a-42ec-11ec-b389-10604b9b4de0  ONLINE      0    0    0      -      -        -        -    -    -    -    -      -      -      -            -      -      -      -
gptid/4fcaf984-42ec-11ec-b389-10604b9b4de0  ONLINE      0    0    0      -      -        -        -    -    -    -    -      -      -      -            -      -      -      -
mirror-3                                      ONLINE      0    0    0
gptid/4f2fc015-42ec-11ec-b389-10604b9b4de0  ONLINE      0    0    0      -      -        -        -    -    -    -    -      -      -      -            -      -      -      -
gptid/506baa26-42ec-11ec-b389-10604b9b4de0  ONLINE      0    0    0      -      -        -        -    -    -    -    -      -      -      -            -      -      -      -
mirror-4                                      ONLINE      0    0    0
gptid/3d2f082b-8113-11ec-90d0-10604b9b4de0  ONLINE      0    0    0      -      -        -        -    -    -    -    -      -      -      -            -      -      -      -
gptid/3d3c7602-8113-11ec-90d0-10604b9b4de0  ONLINE      0    0    0      -      -        -        -    -    -    -    -      -      -      -            -      -      -      -
mirror-5                                      ONLINE      0    0    0
gptid/153e8a02-813c-11ec-90d0-10604b9b4de0  ONLINE      0    0    0      -      -        -        -    -    -    -    -      -      -      -            -      -      -      -
gptid/155e4478-813c-11ec-90d0-10604b9b4de0  ONLINE      0    0    0      -      -        -        -    -    -    -    -      -      -      -            -      -      -      -
mirror-6                                      ONLINE      0    0    0
gptid/15379704-813c-11ec-90d0-10604b9b4de0  ONLINE      0    0    0      -      -        -        -    -    -    -    -      -      -      -            -      -      -      -
gptid/156d8a0a-813c-11ec-90d0-10604b9b4de0  ONLINE      0    0    0      -      -        -        -    -    -    -    -      -      -      -            -      -      -      -
mirror-7                                      ONLINE      0    0    0
gptid/1592a67f-813c-11ec-90d0-10604b9b4de0  ONLINE      0    0    0      -      -        -        -    -    -    -    -      -      -      -            -      -      -      -
gptid/15567a41-813c-11ec-90d0-10604b9b4de0  ONLINE      0    0    0      -      -        -        -    -    -    -    -      -      -      -            -      -      -      -

errors: No known data errors
```

### Expected result
The requested additional fields contain valid info for each of the corresponding drives

**Comment by rincebrain** [comment]
Yeah, those scripts have been broken on every system I've tried them on for a while.

I kept meaning to figure out why, but I so rarely needed them.

It's not everything, but with a few quick local changes (the model/vendor information would take a fair bit of work for at least my setup, lsblk doesn't seem to have any idea how to get that properly on my system, at a quick glance):
```
 # ZPOOL_SCRIPTS_AS_ROOT=1 zpool status -v -c health,model,hours_on,dmdeps,enc,slot,temp,lsblk,defect,size,model,media,test_status,encdev,label,serial phantasm
  pool: phantasm
 state: ONLINE
status: Some supported features are not enabled on the pool. The pool can
        still be used, but some features are unavailable.
action: Enable all features using 'zpool upgrade'. Once this is done,
        the pool may no longer be accessible by software that does not support
        the features. See zpool-features(5) for details.
  scan: resilvered 2.46M in 00:00:01 with 0 errors on Mon Apr 25 07:36:29 2022
config:

        NAME                                    STATE     READ WRITE CKSUM  health  model  hours_on  enc  slot  temp  size  vendor  defect  media  test_status  encdev     label    serial
        phantasm                                ONLINE       0     0     0
          mirror-0                              ONLINE       0     0     0
            ata-WDC_WD100EMAZ-00WJTA0_JEHVB59Z  ONLINE       0     0     0  PASSED      -     23058    -     -    40  9.1T       -       -      -            -       -  phantasm  JEHVB59Z
            ata-WDC_WD100EMAZ-00WJTA0_2YK8Y1WD  ONLINE       0     0     0  PASSED      -     22901    -     -    40  9.1T       -       -      -            -       -  phantasm  2YK8Y1WD
            ata-WDC_WD100EMAZ-00WJTA0_JEK8PB5Z  ONLINE       0     0     0  PASSED      -     23027    -     -    41  9.1T       -       -      -            -       -  phantasm  JEK8PB5Z
```

```
--- /old/usr/lib/zfs-linux/zpool.d/health     2022-02-07 08:27:14.000000000 -0500
+++ /usr/lib/zfs-linux/zpool.d/health   2022-05-02 19:49:10.325534686 -0400
@@ -69,10 +69,17 @@
         exit
 fi

+
 smartctl_path=$(command -v smartctl)

+if [ ! -b "$VDEV_UPATH" ]; then
+       somepath="${VDEV_PATH}"
+else
+       somepath="${VDEV_UPATH}"
+fi
+
 # shellcheck disable=SC2015
-if [ -b "$VDEV_UPATH" ] && [ -x "$smartctl_path" ] || [ -n "$samples" ] ; then
+if [ -b "$somepath" ] && [ -x "$smartctl_path" ] || [ -n "$samples" ] ; then
        if [ -n "$samples" ] ; then
                # cat a smartctl output text file instead of running smartctl
                # on a vdev (only used for developer testing).
@@ -80,7 +87,7 @@
                echo "file=$file"
                raw_out=$(cat "$samples/$file")
        else
-               raw_out=$(eval "sudo $smartctl_path -a $VDEV_UPATH")
+               raw_out=$(eval "sudo $smartctl_path -a $somepath")
        fi

        # What kind of drive are we?  Look for the right line in smartctl:
@@ -188,7 +195,7 @@
 }

 END {print "type="type; ORS="\n"; print ""}
-');
+' | uniq);
 fi
 type=$(echo "$out" | grep '^type=' | cut -d '=' -f 2)

```

```
--- /monolith/sorted/backups/steamer_slash/usr/lib/zfs-linux/zpool.d/lsblk      2022-02-07 08:27:14.000000000 -0500
+++ /usr/lib/zfs-linux/zpool.d/lsblk    2022-05-02 19:43:18.148783668 -0400
@@ -61,20 +61,27 @@
        list=$(echo "$script" | tr '[:upper:]' '[:lower:]')
 fi

+if [ ! -b "$VDEV_UPATH" ]; then
+       somepath="${VDEV_PATH}"
+else
+       somepath="${VDEV_UPATH}"
+fi
+
+
 # Older versions of lsblk don't support all these values (like SERIAL).
 for i in $list ; do

        # Special case: Looking up the size of a file-based vdev can't
        # be done with lsblk.
-       if [ "$i" = "size" ] && [ -f "$VDEV_UPATH" ] ; then
-               size=$(du -h --apparent-size "$VDEV_UPATH" | cut -f 1)
+       if [ "$i" = "size" ] && [ -f "$somepath" ] ; then
+               size=$(du -h --apparent-size "$somepath" | cut -f 1)
                echo "size=$size"
                continue
        fi


        val=""
-       if val=$(eval "lsblk -dl -n -o $i $VDEV_UPATH 2>/dev/null") ; then
+       if val=$(eval "lsblk -dl -n -o $i $somepath 2>/dev/null") ; then
                # Remove leading/trailing whitespace from value
                val=$(echo "$val" | sed -e 's/^[[:space:]]*//' \
                     -e 's/[[:space:]]*$//')
```

**Comment by behlendorf** [comment]
Closing, this should be resolved by the  #13436 workaround.
